app/app.ts

```diff
@@ -164,10 +168,17 @@ function handleReqObjs(req: express.Request, res: express.Response) {
     return x;
   };
 
-  const mapNameQuery = mapName ? `AND map_name = @map_name` : '';
+  let mapNameQuery = mapName ? `AND map_name = @map_name` : '';
+  if (mapType == 'MainField' && mapName) {
+    if (['Sky', 'Cave', 'DeepHole'].includes(mapName))
+      mapNameQuery = ` AND map_name like '${mapName}__%' `;
```

Comment:
We should use GLOB instead of LIKE here, otherwise this will not use the index on map_name which will make this much slower.

Reply:
also, do not use mapName directly as this introduces a SQL injection vulnerability

Reply:
Done